rkt/pubkey/pubkey.go

```diff
@@ -210,7 +217,13 @@ func downloadKey(u *url.URL, skipTLSCheck bool) (*os.File, error) {
 
 	// TODO(krnowak): we should probably apply credential headers
 	// from config here
-	client := getClient(skipTLSCheck)
+	var client *http.Client
+	if skipTLSCheck {
+		client = insecureClient
```

Comment:
This is now going to have race conditions in parallel scenarios. Can we just revert back to creating instead of re-using? it seems like a premature optimisation to me.